src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -3132,6 +3141,7 @@ public void onSelection(SelectionEvent<Integer> event)
       if (event.getSelectedItem() >= 0)
       {
          activeEditor_ = editors_.get(event.getSelectedItem());
+         consoleEditorHadFocusLast_ = false;
```

Comment:
I think it's fine to leave it. The issue @wch pointed out occurs only when you attempt to use an interactive addin when there is no source document available (ie, only the console is open, and the source window is hidden)